src/Tools/dotnet-monitor/AbstractEventSourcePipeline.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.Monitoring.EventPipe;
+using System;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tracing;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    public abstract class AbstractEventSourcePipeline : IAsyncDisposable
```

Comment:
Chatted with @wiktork, will rename to `EventSourcePipelineBase` to follow the naming pattern we use elsewhere